notification-config-service-api/src/main/proto/org/hypertrace/notification/config/service/v1/notification_rule.proto

```diff
@@ -0,0 +1,25 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+
+package org.hypertrace.notification.config.service.v1;
+
+message NewNotificationRule {
+  NotificationRuleMutableData notification_rule_data = 1;
+}
+
+message NotificationRule {
+  string id = 1;
+  NotificationRuleMutableData notification_rule_data = 2;
+}
+
+message NotificationRuleMutableData {
+  string rule_name = 1;
+  string description = 2;
+
+  string event_condition_id = 3;
```

Comment:
is every notification rule associated with exactly one event condition? Or could it it have multiple? e.g. X and Y

Reply:
Good point. 

These are independent events. So, for a rule - send a notification if both event X and event Y occurred - notifier will need more information for evaluation (How long should wait for both events to occur? During what time-window should it consider for evaluation?). 

Instead of going via notifier, as the second way, we can consider such rules as a new event condition type with its evaluator.  So, I think we can go with one-one mapping.

Reply:
Yeah makes sense. I was thinking of cases that multiple conditions apply to the same type of event, but those likely belong to a single top level condition that may contain multiple predicates (e.g. a metric that has name X, has violated a baseline Y and belongs to API Z).